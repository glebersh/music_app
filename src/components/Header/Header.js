import React from 'react';
import Search from '../Search';

const Header = () => {
  return (
    <header>
      <h1>AppLOGO</h1>
      <Search />
    </header>
  )
};

export default Header;
