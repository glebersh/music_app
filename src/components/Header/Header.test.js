import { render } from "@testing-library/react";
import Header from "./Header";
import '@testing-library/jest-dom';

import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";

import * as reduxHooks from 'react-redux';
jest.mock('react-redux');
const mockedUseDispatch = jest.spyOn(reduxHooks, 'useDispatch');

describe('Header test', () => {
  it('Should match snapshot', () => {
    const dispatch = jest.fn();
    mockedUseDispatch.mockReturnValue(dispatch);
    const header = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>);
    expect(header).toMatchSnapshot();
  });

  it('Should close alert on click event', () => {
    const dispatch = jest.fn();
    mockedUseDispatch.mockReturnValue(dispatch);
    const header = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>);
    userEvent.click(header.getByRole('alert-close-button'));
    expect(header.getByTestId('alert-toast')).not.toBeVisible();
  });
});