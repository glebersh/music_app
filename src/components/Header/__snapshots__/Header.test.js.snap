// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-alert css-6r3282"
        data-testid="alert-toast"
        id="token-alert"
        role="alert"
      >
        <span
          class="chakra-alert__icon css-1c6vscd"
        >
          <svg
            class="chakra-icon css-fi0ww0"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
              fill="currentColor"
            />
          </svg>
        </span>
        Keep in mind, that access token expire time is 1 hour, after this time you will be redirected to login page!
        <button
          aria-label="Close"
          class="css-1b5eo55"
          role="alert-close-button"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="css-1jzcfnc"
      >
        <p
          class="chakra-text css-9br895"
        >
          AppLogo
        </p>
        <div
          class="css-8atqhb"
        >
          <form>
            <div
              class="css-zqq8sv"
            >
              <input
                class="chakra-input css-15u1lqs"
                role="search"
                type="search"
              />
              <input
                class="chakra-input css-hn1347"
                role="search-submit-button"
                type="submit"
                value="Search"
              />
            </div>
            <div
              class="css-1hs5feh"
            >
              <span
                class="css-13tvnpb"
                role="search-filter-button"
              >
                Multi
              </span>
              <span
                class="css-13tvnpb"
                role="search-filter-button"
              >
                Songs
              </span>
              <span
                class="css-13tvnpb"
                role="search-filter-button"
              >
                Albums
              </span>
              <span
                class="css-13tvnpb"
                role="search-filter-button"
              >
                Artists
              </span>
              <span
                class="css-1qcgofx"
                role="search-filter-button"
              >
                Playlists
              </span>
            </div>
          </form>
        </div>
        <button
          class="chakra-button css-mol7tl"
          type="button"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            >
              <circle
                cx="12"
                cy="12"
                r="5"
              />
              <path
                d="M12 1v2"
              />
              <path
                d="M12 21v2"
              />
              <path
                d="M4.22 4.22l1.42 1.42"
              />
              <path
                d="M18.36 18.36l1.42 1.42"
              />
              <path
                d="M1 12h2"
              />
              <path
                d="M21 12h2"
              />
              <path
                d="M4.22 19.78l1.42-1.42"
              />
              <path
                d="M18.36 5.64l1.42-1.42"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-alert css-6r3282"
      data-testid="alert-toast"
      id="token-alert"
      role="alert"
    >
      <span
        class="chakra-alert__icon css-1c6vscd"
      >
        <svg
          class="chakra-icon css-fi0ww0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
            fill="currentColor"
          />
        </svg>
      </span>
      Keep in mind, that access token expire time is 1 hour, after this time you will be redirected to login page!
      <button
        aria-label="Close"
        class="css-1b5eo55"
        role="alert-close-button"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-1jzcfnc"
    >
      <p
        class="chakra-text css-9br895"
      >
        AppLogo
      </p>
      <div
        class="css-8atqhb"
      >
        <form>
          <div
            class="css-zqq8sv"
          >
            <input
              class="chakra-input css-15u1lqs"
              role="search"
              type="search"
            />
            <input
              class="chakra-input css-hn1347"
              role="search-submit-button"
              type="submit"
              value="Search"
            />
          </div>
          <div
            class="css-1hs5feh"
          >
            <span
              class="css-13tvnpb"
              role="search-filter-button"
            >
              Multi
            </span>
            <span
              class="css-13tvnpb"
              role="search-filter-button"
            >
              Songs
            </span>
            <span
              class="css-13tvnpb"
              role="search-filter-button"
            >
              Albums
            </span>
            <span
              class="css-13tvnpb"
              role="search-filter-button"
            >
              Artists
            </span>
            <span
              class="css-1qcgofx"
              role="search-filter-button"
            >
              Playlists
            </span>
          </div>
        </form>
      </div>
      <button
        class="chakra-button css-mol7tl"
        type="button"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          >
            <circle
              cx="12"
              cy="12"
              r="5"
            />
            <path
              d="M12 1v2"
            />
            <path
              d="M12 21v2"
            />
            <path
              d="M4.22 4.22l1.42 1.42"
            />
            <path
              d="M18.36 18.36l1.42 1.42"
            />
            <path
              d="M1 12h2"
            />
            <path
              d="M21 12h2"
            />
            <path
              d="M4.22 19.78l1.42-1.42"
            />
            <path
              d="M18.36 5.64l1.42-1.42"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
